### PR TITLE
we want to use vmware fusion and virtualbox as a vagrant provider

### PR DESCRIPTION
**Pull request by mmailand**
```
add support for vagrant provider virtualbox

    * 05_up.sh detects installed provider and defaults vmware_fusion
    * Vagrantfile add nic, memory and vcpu customizations for virtualbox
    * examples/common.yaml change openstack:nova:libvirt_type to qemu
```

### CODE REVIEW

File: examples/allinone/05_up.sh [hunk 1]
```diff
@@ -1,4 +1,26 @@
...
+    then
+        echo "vmware_fusion"
+    elif [[ -x `which VirtualBox` ]]
+    then
+        echo "virtualbox"
+    else
+        exit 1
+    fi
+}   # ----------  end of function provider  ----------
+
```
Comment (cmurphy) [inline examples/allinone/05_up.sh]: The provider() function doesn't use the $PROVIDER variable that you set above. Shouldn't it try to use $PROVIDER first if it is set by the user?

File: examples/allinone/Vagrantfile [hunk 2]
```diff
@@ -49,6 +54,13 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
         v.vmx["numvcpus"] = "4"
         v.vmx["vhv.enable"] = "TRUE"
     end
+    allinone.vm.provider "virtualbox" do |vb|
+        vb.customize ["modifyvm", :id, "--nictype1", "virtio"]
+        vb.customize ["modifyvm", :id, "--nicpromisc3", "allow-all"]
+        vb.cpus = 4
+        vb.memory = 6144
+		vb.customize ["modifyvm", :id, "--natdnspassdomain1", "off"]
```
Comment (cmurphy) [inline examples/allinone/Vagrantfile]: whitespace issues here

**Comment by cmurphy** [comment]
@saneax could you take a look too, since you were working on something similar?

File: examples/common.yaml [hunk 3]
```diff
@@ -132,7 +132,7 @@ openstack::swift::hash_suffix: 'pop-bang'
 
 ######## Nova
 
-openstack::nova::libvirt_type: 'kvm'
+openstack::nova::libvirt_type: 'qemu'
```
Comment (cmurphy) [inline examples/common.yaml]: This change needs to be made in examples/allinone.yaml as well

Comment (saneax) [inline examples/common.yaml]: Greetings,

I will take this up.

One question although, should this work go into puppetlabs/openstack or
stackforge/openstack.
Being new I am still a bit confused on that.

regards
/sanjay

On Wed, Jun 10, 2015 at 3:17 AM, Colleen Murphy <notifications@github.com>
wrote:

> In examples/common.yaml
> <https://github.com/puppetlabs/puppetlabs-openstack/pull/199#discussion_r32068357>
> :
>
> > @@ -132,7 +132,7 @@ openstack::swift::hash_suffix: 'pop-bang'
> >
> >  ######## Nova
> >
> > -openstack::nova::libvirt_type: 'kvm'
> > +openstack::nova::libvirt_type: 'qemu'
>
> This change needs to be made in examples/allinone.yaml as well
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppetlabs/puppetlabs-openstack/pull/199/files#r32068357>
> .
>



-- 
Sanjay Upadhyay
http://saneax.blogspot.com

Comment (cmurphy) [inline examples/common.yaml]: The puppet-openstack module on stackforge is deprecated.

There is also stackforge/puppet-openstack-cloud but it does not have any vagrant examples so this change would not be applicable.

Comment (saneax) [inline examples/common.yaml]: Thanks Colleen for the clarification, I will work on this and resubmit the
patch in a week.

I hope that's an acceptable timeline :)

Moreover, I would like some mentoring so I will seek help on IRC.

regards
/sanjay

On Thu, Jun 11, 2015 at 7:15 AM, Colleen Murphy <notifications@github.com>
wrote:

> In examples/common.yaml
> <https://github.com/puppetlabs/puppetlabs-openstack/pull/199#discussion_r32183988>
> :
>
> > @@ -132,7 +132,7 @@ openstack::swift::hash_suffix: 'pop-bang'
> >
> >  ######## Nova
> >
> > -openstack::nova::libvirt_type: 'kvm'
> > +openstack::nova::libvirt_type: 'qemu'
>
> The puppet-openstack module on stackforge is deprecated.
>
> There is also stackforge/puppet-openstack-cloud but it does not have any
> vagrant examples so this change would not be applicable.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppetlabs/puppetlabs-openstack/pull/199/files#r32183988>
> .
>



-- 
Sanjay Upadhyay
http://saneax.blogspot.com

**Comment by cmurphy** [comment]
Any update on this?

**Comment by mmailand** [comment]
@cmurphy I could push a new branch with fixed the issue with checking the env for the provider type and with fixed whitespacing, would that be okay?

**Comment by cmurphy** [comment]
@mmailand plese do, thanks!

**Comment by saneax** [comment]
Folks,

sorry not to update so far, Will try to get this completed over the
weekend. However whatever is quicker and better is perfectly OK.

warm regards
/sanjay

On Thu, Jul 16, 2015 at 6:46 AM, Colleen Murphy <notifications@github.com>
wrote:

> @mmailand <https://github.com/mmailand> plese do, thanks!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppetlabs/puppetlabs-openstack/pull/199#issuecomment-121794961>
> .
>



-- 
Sanjay Upadhyay
http://saneax.blogspot.com

**Pull request closed without merging**